lib/factorial.rb

```diff
@@ -1,4 +1,25 @@
 # Computes factorial of the input number and returns it
 def factorial(number)
-  raise NotImplementedError
+
+  if number == nil
+    raise ArgumentError.new("Not valid")
+  elsif number == 0
+    return 1
+  else
+    counter = number
+
+    until counter == 1
+      number = number * decrament(counter)
```

Comment:
Minor: it seems to be an overkill to create a method to just subtract one from counter when you need to do it in any case later in this block. Couldn't lines 12 and 13 simply be changed to:
```ruby
 counter -= 1
 number = number * counter
```